### ISSUE TITLE
[Feature Request] Addition of Advanced hardsuits to be creatable by Roboticists, namely a Flight Exosuit.

### ISSUE THREAD

**Issue by Megadeadly** [d3athrow/vgstation13#30125]
First time doing anything on Github so apologies if I bungle anything.

Reason why: Personally, cause I loved the hell out of them on a now dead server. Otherwise; As far as I know, non of the hard suits are unique in terms of function. Simply different armor bonuses and resistances, with potentially some extra goodies tacked on. The flightsuit on the hand, is unique in that its got a momentum based system flight system, and integrated jet pack, with extra abilities like binocular vision. In effect, it is the ultimate jetpack hard suit combo.

How its balanced: Gating the suit behind phazon would be reasonable I think, and prevent it from being a round constant without support such as xenobiology, or at least quickly obtainable in a rush most of the time. While the extreme speeds that you can reach with the suit are strong, deploying the jetpack prevents you from wearing a backpack, and ramming a wall at high speed knocks you down. It would be nearly impossible I think without incredible skill to be able to fly at top speeds and do anything other than flee. And as long as its armor isn't incredibly high, although I think it should have at least a minor amount in all, it won't replace a syndie suit or sec suit for strength.

How it should be added in: I dunno. I don't know how well it will merge, and some things about it will need to be changed, at least the way its made, seeing as we don't have titanium. I'll be useless in that field, as I have no experience messing with code.

Link to the flightsuit file, dunno the right way to do this.
https://github.com/OracleStation/OracleStation/blob/fca5fb13fc3354056fde4b39cb7c8cfec4f839f7/code/modules/clothing/spacesuits/flightsuit.dm

Any feedback on the correct way to format or anything like that, as well as thoughts on this would be appreciated.

TLDR: I think the flight suit is neat, and I don't think it would be game-breaking to add. I hope.
